### ISSUE TITLE
Motivation behind HKDF and GCM

### ISSUE THREAD

**Issue by pedrouid** [normano/ecies_25519#1]
Hi @normano, I just came across your library and I have implemented something very similar myself: https://github.com/pedrouid/ecies-25519

However the main differences are the HKDF and GCM. For my library I have used instead SHA512 for KDF and CBC for AES encryption. I'm curious however about your motivation behind these algorithms

I'm not a cryptographer but I would love to know more about any properties that might have justified these decisions

Thanks in advance 🙏

**Comment by normano** [comment]
I only learned about ECIES recently since I wanted a way to encrypt/decrypt arbitrary data with asymmetric keys. I wasn't expecting to implement this library let alone make it public.

There are several sources about these algorithms where you can read about HKDF and GCM:  https://joyofcryptography.com/, https://www.crypto101.io/

Apple's Secure Enclave uses ECIES and a person helps break it down. (https://darthnull.org/secure-enclave-ecies/).

There are other interesting posts to read about like (https://soatok.blog/2020/05/13/why-aes-gcm-sucks/). AES is only as secure as the entropy given to it.

If I remember correctly there was nothing that SHA-512 offered over SHA-256 given the size of keys worked on on a 64bit machine. This one is interesting https://pycryptodome.readthedocs.io/en/latest/src/hash/sha512.html

tldr?
GCM - For AEAD (https://joyofcryptography.com/pdf/chap12.pdf).
HKDF-SHA-256 - For increasing entropy of the secret entered into AES.

I do not expect anyone other than me to use this library and that's why it is offered under MPL 2.0 as I have no liability for the use of it.
